File: python/app/domain/user_repository.py
from abc import ABC, abstractmethod

from app.domain.user import User


class UserRepository(ABC):
    """用户Repository接口"""

    @abstractmethod
    def insert(self, entity: User) -> None:
        """插入"""
        pass

    @abstractmethod
    def update(self, entity: User) -> None:
        """更新"""
        pass

    @abstractmethod
    def delete_by_id(self, id: str) -> None:
        """根据Id删除"""
        pass

    @abstractmethod
    def select_by_id(self, id: str) -> User | None:
        """根据Id查询，找不到返回null"""
        pass

    @abstractmethod
    def select_by_id_req(self, id: str) -> User:
        """根据Id查询，找不到抛出异常"""
        pass

    @abstractmethod
    def select_by_ids(self, id: list[str]) -> list[User]:
        """根据Id集合查询"""
        pass

    @abstractmethod
    def select_by_username(self, username: str) -> User | None:
        """根据用户名查询，找不到返回null"""
        pass

File: dotnet/src/domain/UserRepository.cs
namespace BackendExample.Domain
{
    /// <summary>
    /// 用户 Repository 接口
    /// </summary>
    public interface UserRepository
    {
        /// <summary>
        /// 插入
        /// </summary>
        Task Insert(User entity);

        /// <summary>
        /// 更新
        /// </summary>
        Task Update(User entity);

        /// <summary>
        /// 根据 id 删除
        /// </summary>
        Task DeleteById(String id);

        /// <summary>
        /// 根据 id 查询，找不到返回 null
        /// </summary>
        Task<User?> SelectById(String id);

        /// <summary>
        /// 根据 id 查询，找不到抛出异常
        /// </summary>
        Task<User> SelectByIdReq(String id);

        /// <summary>
        /// 根据 id 集合查询
        /// </summary>
        Task<IList<User>> SelectByIds(List<String> ids);

        /// <summary>
        /// 根据用户名查询，找不到返回 null
        /// </summary>
        Task<User?> SelectByUsername(String username);
    }
}
